cmd/root.go

```diff
@@ -81,24 +88,31 @@ func Execute() {
 	}
 }
 func startScanner(namespace string, clusterWide bool, scanner Scanner) error {
-	if clusterWide {
-		if err := scanner.ScanClusterWideResources(); err != nil {
-			return err
-		}
-	}
-
-	if namespace != "" {
-		if err := scanner.ScanNamespace(namespace); err != nil {
-			return err
+	switch clusterWide {
+	case true:
+		if namespace != "" {
+			log.Fatal().Msg("Cannot scan cluster wide and only a namespace at the same time")
+		} else {
+			if err := scanner.ScanClusterWideResources(); err != nil {
+				return err
+			}
 		}
-	} else if !clusterWide {
-		// FIXME ScanAllNamespaces is not implemented. Therefore, if we are
-		// scanning cluster wide resource do not trigger this scan. It will failed anyway
-		if err := scanner.ScanAllNamespaces(); err != nil {
-			return err
+	case false:
+		if namespace != "" {
+			if err := scanner.ScanNamespace(namespace); err != nil {
+				return err
+			}
+		} else {
+			// neither clusterWide flag nor namespace was provided, default
+			// behaviour of scanning cluster wide and all ns
+			if err := scanner.ScanClusterWideResources(); err != nil {
+				return err
+			}
+			if err := scanner.ScanAllNamespaces(); err != nil {
+				return err
+			}
```

Comment:
My personal preference... I found weird a switch case for a boolean variable.

Reply:
Me too. But the linters complain 3 nested ifs are too much..

Reply:
what about something like that:

```golang
func scanNamespaces(nsNames []string) error {
   nsWithErrors := []string{}
   for namespace in nsNames {
     if err : = scanner.ScanNamespace(namespace); err != nil {
        log.Error(err)
        nsWithErrors = append(nsWithErrors, namespace)
     }
   }
   if somethingWentWrong {
     return fmt.Errorf("Something went wrong while scanning these namespaces, checks the logs for more details: %s", strings.join(nsWithErrors, ", "))
   }
   return nil
}
```

Then we could have the code do something like that:

```
if cli_flags.clusterwide && cli_flags.namespace != "" {
   return error
}

if cli_flags.clusterwide {
   scan_cluster_wide
}

var nsToScan []string
if cli_flags.namespace != "" {
   nsToScan = []string{cli_flags.namespace}
} else {
   nsToScan = find_all_ns()
}

scanNamespaces(nsToScan)
```

Reply:
that is indeed more elegant, but there, `find_all_ns()` is right now implemented by `s.policiesFetcher.GetAuditedNamespaces()`. I can consume `policiesFetcher` from the RunE function, but that also adds it to the mock tests.

Reply:
refactored even simpler in https://github.com/kubewarden/audit-scanner/pull/53/commits/33ab16402bdcd660536977dddcc338b36222fdc7.